### PR TITLE
[13.0][IMP] sale_stock: Backport of the _compute_qty_to_deliver function from v14.

### PR DESCRIPTION
**Pull request by victoralmau**
Backport of the `_compute_qty_to_deliver()` function from v14.

Commits: https://github.com/odoo/odoo/commit/5e5de2b2b64fd8cb259e72fc1a97dc754da871ad (partial) + https://github.com/odoo/odoo/commit/ea798c07d31fd186af4bf851966f791a96cb13fd

Please @pedrobaeza and @davidvidal can you review it?

@Tecnativa TT32530

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you rescue the same commit that makes this change? I think there are several of them included, counting also with some bugfixes.

**Comment by victoralmau** [comment]
I have rescued the commits related to this change.

**Comment by pedrobaeza** [comment]
cc @chienandalu

**Review by pedrobaeza — APPROVED** [review]

File: addons/sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -215,13 +215,16 @@ class SaleOrderLine(models.Model):
     is_mto = fields.Boolean(compute='_compute_is_mto')
     display_qty_widget = fields.Boolean(compute='_compute_qty_to_deliver')
 
-    @api.depends('product_id', 'product_uom_qty', 'qty_delivered', 'state', 'product_uom')
+    @api.depends('product_type', 'product_uom_qty', 'qty_delivered', 'state', 'move_ids', 'product_uom')
```
Comment (chienandalu) [inline addons/sale_stock/models/sale_order.py:218]: As we saw recently at least in v13 related fields don't behave well in computes, maybe we could leave it as it was or draw the full relation

Comment (pedrobaeza) [reply addons/sale_stock/models/sale_order.py:218]: But this doesn't happen on `depends`. Take into account that this is working in 14.0, that uses the same ORM.

Comment (chienandalu) [reply addons/sale_stock/models/sale_order.py:218]: Ok, then

**Review by chienandalu — APPROVED** [review]

**Comment by victoralmau** [comment]
Finally, it's ready to merge, right?

**Pull request merged**